scipy/spatial/transform/rotation.py

```diff
@@ -0,0 +1,868 @@
+    return product
+
+
+def _elementary_quat_compose(seq, angles, intrinsic=False):
+    result = _make_elementary_quat(seq[0], angles[:, 0])
+
+    for idx, axis in enumerate(seq[1:], start=1):
+        if intrinsic:
+            result = _compose_quat(
+                result,
+                _make_elementary_quat(axis, angles[:, idx]))
+        else:
+            result = _compose_quat(
+                _make_elementary_quat(axis, angles[:, idx]),
+                result)
+    return result
+
+
+class Rotation(object):
+    """Rotation in 3 dimensions.
+
+    This class provides an interface to initialize from and represent rotations
+    with:
+
+        - Quaternions
+        - Direction Cosine Matrices
+        - Rotation Vectors
+        - Euler angles
+
+    The following operations on rotations are supported:
+
+        - Application on vectors
+        - Rotation Composition
+        - Rotation Inversion
+        - Rotation Indexing
+
+    Indexing within a rotation is supported since multiple rotation transforms
+    can be stored within a single `Rotation` instance.
+
+    To create `Rotation` objects use `from_...` classmethods, `__init__` is not
+    supposed to be used directly.
+
+    Methods
+    -------
+    __len__
+    from_quat
+    from_dcm
+    from_rotvec
+    from_euler
+    as_quat
+    as_dcm
+    as_rotvec
+    as_euler
+    apply
+    __mul__
+    inv
+    __getitem__
+    """
+    def __init__(self, quat, normalized=False, copy=True):
+        self._single = False
+        quat = np.asarray(quat, dtype=float)
+
+        if quat.ndim not in [1, 2] or quat.shape[-1] != 4:
+            raise ValueError("Expected `quat` to have shape (4,) or (N x 4), "
+                             "got {}.".format(quat.shape))
+
+        # If a single quaternion is given, convert it to a 2D 1 x 4 matrix but
+        # set self._single to True so that we can return appropriate objects
+        # in the `to_...` methods
+        if quat.shape == (4,):
+            quat = quat[None, :]
+            self._single = True
+
+        if normalized:
+            self._quat = quat.copy() if copy else quat
+        else:
+            self._quat = quat.copy()
+            norms = scipy.linalg.norm(quat, axis=1)
+
+            zero_norms = norms == 0
+            if zero_norms.any():
+                raise ValueError("Found zero norm quaternions in `quat`.")
+
+            # Ensure norm is broadcasted along each column.
+            self._quat[~zero_norms] /= norms[~zero_norms][:, None]
+
+    def __len__(self):
+        """Number of rotations contained in this object.
+
+        Multiple rotations can be stored in a single instance.
+
+        Returns
+        -------
+        length : int
+            Number of rotations stored in object.
+        """
+        return self._quat.shape[0]
+
+    @classmethod
+    def from_quat(cls, quat, normalized=False):
+        """Initialize Rotation from quaternions.
```

Comment:
I think you can omit "Rotation" in similar places. That is "Initialize from quaternions".